### ISSUE TITLE
username restriction

### ISSUE THREAD

**Issue by codeandfix** [minetest/minetest#10964]
## Problem

usernames are very unrestricted. normal humans tend to be case insensitive when talking about usernames.
actually its very easy to spoof other playernames.

## Solutions

a) restrict playernames/usernames to a min/max length def: 3-10 would be fine
b) restrict playernames, so that they are case insensitive 
c) restrict playernames to [A-Z],[a-z],[0-9] only. there is no need to support special chars and unicode in the world of unique ids. (if not already done)
d) add a blocklist of unwanted playernames like: lamer, noob, admin, administrator, root, ...
e) add a list of "premium" nicknames which a reserved for special users. all other have to use not-so-nice-names
    e.1.) do so by limiting normal users to usernames 6-10 chars and premium to 4-6
    e.2.) do so by adding a "whitelist" for premium only
    e.3.) do so by forcing an admin to confirm new nicknames. admin may choose: accept, deny, rename

## Alternatives

block new users to register an already used playername if just the case (up/down) is different.
so a registered user name: SirX blocks Sirx and sirx and so on.

**Comment by rubenwardy** [comment]
> a) restrict playernames/usernames to a min/max length def: 3-10 would be fine

Usernames are limited to 18 characters

> b) restrict playernames, so that they are case insensitive

They are already case insensitive

> c) restrict playernames to [A-Z],[a-z],[0-9] only. there is no need to support special chars and unicode in the world of unique ids. (if not already done)

I believe it's currently `[A-Za-z0-9_]+`

> d) add a blocklist of unwanted playernames like: lamer, noob, admin, administrator, root, ...

This can already be done by server mods, I think it's best to leave it to them

> e) add a list of "premium" nicknames which a reserved for special users. all other have to use not-so-nice-names
> e.1.) do so by limiting normal users to usernames 6-10 chars and premium to 4-6
> e.2.) do so by adding a "whitelist" for premium only

You can do this using mods. I don't see a need to do this, we don't and won't have a premium mode

> e.2.) do so by adding a "whitelist" for premium only
> e.3.) do so by forcing an admin to confirm new nicknames. admin may choose: accept, deny, rename

Eh, this can already be done by a mod and I don't see much value of this in core

Do I sound like a broken record?

**Comment by codeandfix** [comment]
sound like a broken sword?

b)
if usernames are already case-insensitive, then my database is inconsistent after conversion.
i converted it from auth.txt to postgresql and now i have Test and test inside as before, so i thought it is still not case insensitive.
this should be blocked by database and server.

btw: premium means not "payed" but just for good/fine players instead of the "chaos-kids"

**Comment by rubenwardy** [comment]
> if usernames are already case-insensitive, then my database is inconsistent after conversion.
> i converted it from auth.txt to postgresql and now i have Test and test inside as before, so i thought it is still not case insensitive.
> this should be blocked by database and server.

Yeah, this will be a bug then with the postgreSQL auth backend

**Comment by 0-afflatus** [comment]
The majority of this can and should be left to server mods, because this behaviour has to be customisable.

**Comment by rubenwardy** [comment]
I suggest either:

* Making this issue just about the postgreSQL username bug
* OR closing and making a new issue about that bug

**Comment by sorcerykid** [comment]
Most of these requests are best left to mods, as Rubenwardy pointed out. But I can't quite wrap my head around what qualifies as a "special user"? How would you determine that a user is special before they've registered for an account?
